### ISSUE TITLE
HTML printing "remembers' size after size is unchecked in sample.html

### ISSUE THREAD

**Issue by lite1979** [qzind/tray#659]
Ubuntu 20.04, Adopt Open JDK 11, Hotspot, QZ-Tray 2.1.0+62

scaleContent: 'true' for the following test

Steps to reproduce: Print HTML without specifying size to any printer, in my case the CUPS PDF printer or my Brother HL-L2380DW, both of which have US Letter as "paper size".

Specify size, preferably smaller than US Letter if that was the first size for the sake of making comparisons, and print to the same printer.

Uncheck size, and print to this printer again. 

It has the same results as the print job with size defined.

This gets more troublesome if I specify a 4x6 size to a print job on one printer, then un-check the size parameter and send a print job to ANOTHER printer, because the size parameter is still remembered.

Logs:

https://gist.github.com/lite1979/8a0678abd1bf6b57e8824248d62cb752

**Comment by tresf** [comment]
Reproduced on MacOS.  It appears it's because @bberenz used `undefined` for a placeholder and `config.reconfigure(...)` ignores `undefined` values.  I'll see if I can patch it.

**Comment by lite1979** [comment]
For what it's worth, QZ-Tray "remembers" the size when switching to another form of pixel printing as well, i.e. PDF or Image.

**Comment by tresf** [comment]
@lite1979 thanks for reporting.  Should be fixed.  For now you can change the two lines in your sample.html:

```diff
    function updateConfig(cleanConditions) {
-       var pxlSize = undefined;
+       var pxlSize = null;
        if (isChecked($("#pxlSizeActive"), cleanConditions['pxlSizeActive'])) {
            pxlSize = {
                width: $("#pxlSizeWidth").val(),
                height: $("#pxlSizeHeight").val()
            };
        }

-       var pxlBounds = undefined;
+       var pxlBounds = null;
        if (isChecked($("#pxlBoundsActive"), cleanConditions['pxlBoundsActive'])) {
            pxlBounds = {
                x: $("#pxlBoundX").val(),
```

**Comment by lite1979** [comment]
Digging a little deeper, it seems that in order for scaleContent to honor its intention, dimensions need to be swapped for height and width for proper results if a size was specified beforehand.

**Comment by tresf** [comment]
> Digging a little deeper, it seems that in order for scaleContent to honor its intention, dimensions need to be swapped for height and width for proper results if a size was specified beforehand.

Right, because normally we'd size to the paper size.  Now that it's patched, these nuances should disappear.  Let me know if they don't, we might need to track it separately.

**Comment by lite1979** [comment]
This behaves as expected now. Thank you!
